### PR TITLE
feat(aws-lambda-dynamodb) added log group props for lambda-dynamodb

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #316 , if available:*

*Description of changes:*
-Added existingLogGroup prop in buildLogGroup helper function
-Created tests for existing log group and created props
-Updated tests snapshots

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 5362949666495fae15be73f5ebf5d19eafb696cf
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7347d68f-c7a6-4f54-b047-83c74ed9d0d5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: d1700979ac8d353a3b44a05a495c75fe9b375df3
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=10105ada-0b6e-4c33-806e-f68258b784cf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md [hunk 1]
```diff
@@ -55,21 +55,24 @@ _Parameters_
...
 |deployVpc?|`boolean`|Whether to create a new VPC based on `vpcProps` into which to deploy this pattern. Setting this to true will deploy the minimal, most private VPC to run the pattern:<ul><li> One isolated subnet in each Availability Zone used by the CDK program</li><li>`enableDnsHostnames` and `enableDnsSupport` will both be set to true</li></ul>If this property is `true` then `existingVpc` cannot be specified. Defaults to `false`.|
+|existingLogGroup?|[`logs.LogGroup`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-logs.LogGroup.html)|Existing instance of a Log Group object.|
+|logGroupProps?|[`logs.LogGroupProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-logs.LogGroupProps.html)|Optional user-provided properties to override the default properties for the new Log Group. `encryptionKey`, `logGroupName`, `removalPolicy` and `retention` are set by the pattern, so any values for those properties supplied here will be overrriden. If properties are empty, it will deploy with the default properties of a default master key, generated log group name, retain removal policy, and two year retention period.|
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md:66]: This is their chance to override encryptionKey, logGroupName and removalPolicy. The only time we override settings from the client is if the operation of the construct requires certain behavior (eg - endpoints require the DNS settings to be on or they won't work). The issue driving this PR specifically  wanting to set the retention policy, and our tests will probably set the removalPolicy.

File: source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md [hunk 2]
```diff
@@ -55,21 +55,24 @@ _Parameters_
...
 |vpcProps?|[`ec2.VpcProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.VpcProps.html)|Optional user-provided properties to override the default properties for the new VPC. `enableDnsHostnames`, `enableDnsSupport`, `natGateways` and `subnetConfiguration` are set by the pattern, so any values for those properties supplied here will be overrriden. If `deployVpc` is not `true` then this property will be ignored.|
 |deployVpc?|`boolean`|Whether to create a new VPC based on `vpcProps` into which to deploy this pattern. Setting this to true will deploy the minimal, most private VPC to run the pattern:<ul><li> One isolated subnet in each Availability Zone used by the CDK program</li><li>`enableDnsHostnames` and `enableDnsSupport` will both be set to true</li></ul>If this property is `true` then `existingVpc` cannot be specified. Defaults to `false`.|
+|existingLogGroup?|[`logs.LogGroup`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-logs.LogGroup.html)|Existing instance of a Log Group object.|
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md:65]: I think we need to clearly state what log group we're discussing - 

An existing Log Group object to assign to the Lambda function.

or below:

Optional user-provided props to configure the Log Group object for the Lambda function.

**Review by biffgaut — CHANGES_REQUESTED** [review]
Stopped the review after reading the docs, because I think there are behavioral changes required that will change a lot of the rest of the code.

**Comment by biffgaut** [comment]
What we're trying to do here can't be accomplished, see linked Issue #316

**Pull request closed without merging**